service/src/main/java/bio/terra/workspace/app/controller/shared/PropertiesUtils.java

```diff
@@ -28,4 +34,16 @@ public static ApiProperties convertMapToApiProperties(Map<String, String> proper
     properties.forEach((key, value) -> apiProperties.add(new ApiProperty().key(key).value(value)));
     return apiProperties;
   }
+
+  /**
+   * Clear certain properties in the hashmap before making a clone.
+   *
+   * <p>For example, TERRA_WORKSPACE_FOLDER_ID is a workspace specific properties. It needs to be
+   * cleared because it is meaningless in a new workspace.
+   */
+  public static ImmutableMap<String, String> clearSomeProperties(Map<String, String> properties) {
```

Comment:
this method name should indicate this is clearing properties relevant to cloning.
it's also slightly confusing to name it `clear___` given that it's really returning a copy without modifying the input. 

maybe something like `clonableProperties` or `getClonableProperties` instead?

Reply:
it does modify the input. i modified comment and method name to make it clearer.